### DISCUSSION TITLE
Val Loss: inf with train_MaskRCNN_model

### DISCUSSION THREAD

**Discussion by datagistips** [opengeos/geoai#334] [Q&A]
Hi,

Is it normal that train_maskRCNN always gives `Inf` for `Val Loss` ?

`Epoch 1/100: Train Loss: 0.5388, Val Loss: inf, Val IoU: 0.4101
`

It's also the case with the opengeoai [Train Car Detection example](https://opengeoai.org/examples/train_car_detection/)

Here is my output in my case (parking spot detection with personal data, only 2 epochs printed for convenience) :
```
2025-10-18 13:51:24.971305: I tensorflow/core/platform/cpu_feature_guard.cc:210] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 AVX512F FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2025-10-18 13:51:30.503934
100 époques...
datasets/04 global/chips_512x256/images
datasets/04 global/chips_512x256/masks
Using device: cuda
Found 9225 image files and 9225 label files
Training on 7380 images, validating on 1845 images
Epoch: 1, Batch: 1/1845, Loss: 3.4110, Time: 2.23s
Epoch: 1, Batch: 51/1845, Loss: 0.9421, Time: 50.25s
Epoch: 1, Batch: 101/1845, Loss: 0.9528, Time: 52.10s
Epoch: 1, Batch: 151/1845, Loss: 0.8174, Time: 52.47s
Epoch: 1, Batch: 201/1845, Loss: 0.0146, Time: 52.69s
Epoch: 1, Batch: 251/1845, Loss: 1.1226, Time: 52.49s
Epoch: 1, Batch: 301/1845, Loss: 0.6443, Time: 52.01s
Epoch: 1, Batch: 351/1845, Loss: 0.8146, Time: 52.00s
Epoch: 1, Batch: 401/1845, Loss: 0.0170, Time: 51.49s
Epoch: 1, Batch: 451/1845, Loss: 0.6785, Time: 51.73s
Epoch: 1, Batch: 501/1845, Loss: 0.6142, Time: 51.78s
Epoch: 1, Batch: 551/1845, Loss: 0.5175, Time: 52.04s
Epoch: 1, Batch: 601/1845, Loss: 0.3774, Time: 51.94s
Epoch: 1, Batch: 651/1845, Loss: 0.3189, Time: 52.02s
Epoch: 1, Batch: 701/1845, Loss: 0.6197, Time: 51.89s
Epoch: 1, Batch: 751/1845, Loss: 0.9057, Time: 52.10s
Epoch: 1, Batch: 801/1845, Loss: 0.6874, Time: 51.97s
Epoch: 1, Batch: 851/1845, Loss: 0.5398, Time: 52.06s
Epoch: 1, Batch: 901/1845, Loss: 0.4360, Time: 52.11s
Epoch: 1, Batch: 951/1845, Loss: 0.8096, Time: 51.93s
Epoch: 1, Batch: 1001/1845, Loss: 0.6291, Time: 52.08s
Epoch: 1, Batch: 1051/1845, Loss: 0.5465, Time: 52.37s
Epoch: 1, Batch: 1101/1845, Loss: 0.5512, Time: 52.14s
Epoch: 1, Batch: 1151/1845, Loss: 0.6068, Time: 52.27s
Epoch: 1, Batch: 1201/1845, Loss: 0.5478, Time: 52.54s
Epoch: 1, Batch: 1251/1845, Loss: 0.6352, Time: 52.34s
Epoch: 1, Batch: 1301/1845, Loss: 0.7471, Time: 52.15s
Epoch: 1, Batch: 1351/1845, Loss: 0.0082, Time: 51.91s
Epoch: 1, Batch: 1401/1845, Loss: 0.6976, Time: 52.32s
Epoch: 1, Batch: 1451/1845, Loss: 0.6810, Time: 51.89s
Epoch: 1, Batch: 1501/1845, Loss: 0.7122, Time: 52.08s
Epoch: 1, Batch: 1551/1845, Loss: 0.3136, Time: 52.19s
Epoch: 1, Batch: 1601/1845, Loss: 0.6293, Time: 51.92s
Epoch: 1, Batch: 1651/1845, Loss: 0.4142, Time: 52.34s
Epoch: 1, Batch: 1701/1845, Loss: 0.0099, Time: 52.56s
Epoch: 1, Batch: 1751/1845, Loss: 0.6842, Time: 52.45s
Epoch: 1, Batch: 1801/1845, Loss: 0.4574, Time: 52.17s
Epoch 1/100: Train Loss: 0.5388, Val Loss: inf, Val IoU: 0.4101
Saving best model with IoU: 0.4101
Epoch: 2, Batch: 1/1845, Loss: 0.4243, Time: 1.66s
Epoch: 2, Batch: 51/1845, Loss: 0.4680, Time: 52.86s
Epoch: 2, Batch: 101/1845, Loss: 0.6284, Time: 52.98s
Epoch: 2, Batch: 151/1845, Loss: 0.4276, Time: 52.65s
Epoch: 2, Batch: 201/1845, Loss: 0.5481, Time: 52.81s
Epoch: 2, Batch: 251/1845, Loss: 0.5269, Time: 52.77s
Epoch: 2, Batch: 301/1845, Loss: 0.6813, Time: 52.43s
Epoch: 2, Batch: 351/1845, Loss: 0.3074, Time: 52.55s
Epoch: 2, Batch: 401/1845, Loss: 0.2229, Time: 52.69s
Epoch: 2, Batch: 451/1845, Loss: 0.3152, Time: 52.59s
Epoch: 2, Batch: 501/1845, Loss: 0.6851, Time: 52.78s
Epoch: 2, Batch: 551/1845, Loss: 0.6415, Time: 52.86s
Epoch: 2, Batch: 601/1845, Loss: 0.4436, Time: 52.50s
Epoch: 2, Batch: 651/1845, Loss: 0.5853, Time: 52.15s
Epoch: 2, Batch: 701/1845, Loss: 0.3764, Time: 52.49s
Epoch: 2, Batch: 751/1845, Loss: 0.4988, Time: 52.59s
Epoch: 2, Batch: 801/1845, Loss: 0.4663, Time: 52.44s
Epoch: 2, Batch: 851/1845, Loss: 0.7377, Time: 52.28s
Epoch: 2, Batch: 901/1845, Loss: 0.3473, Time: 52.30s
Epoch: 2, Batch: 951/1845, Loss: 0.6191, Time: 52.56s
Epoch: 2, Batch: 1001/1845, Loss: 0.5167, Time: 52.36s
Epoch: 2, Batch: 1051/1845, Loss: 0.3684, Time: 52.18s
Epoch: 2, Batch: 1101/1845, Loss: 0.4192, Time: 52.50s
Epoch: 2, Batch: 1151/1845, Loss: 0.3565, Time: 52.52s
Epoch: 2, Batch: 1201/1845, Loss: 0.2714, Time: 52.22s
Epoch: 2, Batch: 1251/1845, Loss: 0.4116, Time: 52.20s
Epoch: 2, Batch: 1301/1845, Loss: 0.0053, Time: 52.18s
Epoch: 2, Batch: 1351/1845, Loss: 0.6647, Time: 52.22s
Epoch: 2, Batch: 1401/1845, Loss: 0.2387, Time: 52.42s
Epoch: 2, Batch: 1451/1845, Loss: 0.5363, Time: 52.56s
Epoch: 2, Batch: 1501/1845, Loss: 0.2659, Time: 52.21s
Epoch: 2, Batch: 1551/1845, Loss: 0.5260, Time: 52.27s
Epoch: 2, Batch: 1601/1845, Loss: 0.5563, Time: 52.48s
Epoch: 2, Batch: 1651/1845, Loss: 0.3608, Time: 52.02s
Epoch: 2, Batch: 1701/1845, Loss: 0.3983, Time: 52.18s
Epoch: 2, Batch: 1751/1845, Loss: 0.5102, Time: 51.90s
Epoch: 2, Batch: 1801/1845, Loss: 0.5245, Time: 52.44s
Epoch 2/100: Train Loss: 0.4288, Val Loss: inf, Val IoU: 0.4518
Saving best model with IoU: 0.4518
```
